RedacteurPortaal.Api/Controllers/AddressController.cs

```diff
@@ -66,17 +79,25 @@ public async Task<IActionResult> DeleteAddress(Guid id)
         {
             await this.grainService.DeleteGrain(id);
             this.logger.LogInformation("Address deleted successfully");
-            return this.Ok("Address deleted");
+            return this.NoContent();
         }
 
         [HttpPatch]
-        public async Task<ActionResult<AddressDTO>> UpdateAddress(Guid guid,[FromBody]AddressDTO addressDTO)
+        public async Task<ActionResult<AddressDTO>> UpdateAddress(Guid id, [FromBody]UpdateAddressRequest addressDTO)
         {
+            TypeAdapterConfig<UpdateAddressRequest, AddressModel>
+            .NewConfig()
+            .Map(dest => dest.Id,
+                src => id);
+
             var address = addressDTO.Adapt<AddressModel>();
-            var grain = await this.grainService.GetGrain(guid);
+            var grain = await this.grainService.GetGrain(id);
             await grain.UpdateAddress(address);
             this.logger.LogInformation("Address updated succesfully");
-            return this.StatusCode(204, "Address updated");
+            var updatedGrain = await this.grainService.GetGrain(address.Id);
+            var updatedItem = await updatedGrain.Get();
```

Comment:
Not necessary

Reply:
Kun je misschien de Update method aanpassen dat hij de grain returned? dat scheelt heel wat lines in de controller